redgenes/utils.py

```diff
@@ -1,29 +1,138 @@
+import gzip
+import shutil
 import subprocess
+from BCBio import GFF
+from pathlib import Path
+from contextlib import contextmanager
+from redgenes.exceptions import InvalidFna
 
 
-def run_command(commands, errortext):
-    """Basic function to run bash commands using subprocess."""
+################################
+# Generate sql insert statements
+################################
+def generate_insert_stmt(table_name, column_names):
+    stmt = f"insert into {table_name} ({', '.join(column_names)}) values ({', '.join(['?']*len(column_names))});"
```

Comment:
This is only safe if the caller can be absolutely and forever assured to be a trusted entity. If dynamic SQL is required, then it is important to use mechanisms provided by the underlying library to protect against SQL injection (whether intentional or accidental). I don't know off hand what the best approach for sqlite3 is, but for example, psycopg2 for postgres has [SQL literals](https://www.psycopg.org/docs/sql.html). Psycopg2 has [more detail](https://www.psycopg.org/docs/usage.html#the-problem-with-the-query-parameters) on concern and associated links, and please do note the block of text in red :)